Bulk exclude backlogged reportbacks on node 362

chloealee opened the pull request:
#### What's this PR do?

Script to bulk exclude any fids for node 362 that are pending and whose reportback has a quantity of less than 200. 
#### How should this be manually tested?

When ssh'd in your vagrant environment, run `drush updb -y` in `dosomething@dev:/var/www/dev.dosomething.org/html$`

Check `http://dev.dosomething.org:8888/us/node/362/rb/` and only `fids` with a quantity over 200 should still be pending and appear in the inbox. 
#### What are the relevant tickets?

Fixes #6344

chloealee committed:
writes a script to bulk exclude any fids for node 362 that are pending and whose reportback has a quantity of less than 200

angaither commented on lib/modules/dosomething/dosomething_reportback/dosomething_reportback.install:
```diff
@@ -910,3 +910,24 @@ function dosomething_reportback_update_7033() {
+        )
+      )
+      ->condition('fid', $fid->fid)
+      ->execute();
```
I think you'll also want to update the `flagged` status on the reportback to 0 as well

chloealee replied:
@angaither per the third bullet point in [Nami's comment](https://github.com/DoSomething/phoenix/issues/6344#issuecomment-205929056) would this be misleading if other fids associated with this rb are marked differently?

angaither replied:
@chloealee `flagged` is applied to the entire reportback if ONE item is flagged. if 0 items are `flagged` that property is set to `0`

chloealee replied:
@angaither if the `fid` is marked as `excluded`, does that mean the reportback is `flagged`?

angaither replied:
@chloealee no, only if one item is marked as `flagged`

the logic [here](https://github.com/DoSomething/phoenix/blob/7b2a84aa399c9bd2eb9a91ddc6ad60bf50e5ccd1/lib/modules/dosomething/dosomething_reportback/includes/Reportback.php#L356) might help

chloealee replied:
@angaither wait nevermind about the comment above. 

The next step would be to do another query to get all the `fids` associated with the reportback and see if any of those are `flagged` - if so, change the `flagged` status of the reportback to `1` and if not, change the `flagged` status to `0`?

angaither replied:
instead of doing a `db_update` like this you may be able to use the function i linked to above if you load the reportback item

chloealee replied:
@angaither ah thank you!

angaither commented on lib/modules/dosomething/dosomething_reportback/dosomething_reportback.install:
```diff
@@ -910,3 +910,24 @@ function dosomething_reportback_update_7033() {
+ * Bulk excludes backlogged reportbacks on node 362.
+ * Updates any fid with a pending status to be marked as excluded if the quantity in the reportback is less than 200.  
+ */   
+function dosomething_reportback_update_7039() {
```
I think this is skipping a few updates. looks like prod is on `7033`

chloealee replied:
ah yes, forgot to change that back - thanks!

chloealee committed:
changes number in function

angaither commented:
@chloealee also this campaign reopens tomorrow, so you'll want to only target reportbacks with `run_nid = 5439`

mikefantini commented:
@chloealee @angaither there might also be reportbacks with `run_nid = 2341`

chloealee committed:
adds certain run_nids to target and adds logic to change reportback flagged status

chloealee commented:
thanks @angaither and @mikefantini ! Added those `run_nid`s to target along with the logic to change the reportback `flagged` status.

chloealee committed:
changes number in function

angaither commented:
@chloealee how have you tested this? did you pull rbs from a different campaign? I'd love to deploy just this branch to thor to take a look to make sure it does what's expected

chloealee commented:
@angaither yup! I was testing on a different campaign on my local and added a reportback with 2 reportback items. I ran the script and saw that both reportback items statuses were changed to `excluded` and the parent reportback `flagged` status changed to `0`. 

Then, I added two more reportback items and changed one `status` to `flagged`. I ran the script and the second reportback item's status updated to `excluded` and the reportback `flagged` status remained as `1`.

angaither commented:
@chloealee kool, let's deploy this to thor to test the branch with the real content then!
